registry/slices/helpers.go

```diff
@@ -29,8 +50,20 @@ func (sr *SlicesRegistry) isComparable(v any) bool {
 	}
 }
 
-// flattenSlice recursively flattens a slice of slices into a single slice.
-// It is used by the Flatten function.
+// flattenSlice takes a slice or array and recursively flattens it into a
+// single-dimensional list of elements. The remainingDeep parameter controls
+// the maximum depth of the flattening, with -1 indicating infinite depth and
+// 0 indicating no recursion. The function returns a slice of the flattened
+// elements.
+//
+// Parameters:
+//
+//	val - the slice or array to flatten
+//	remainingDeep - the maximum depth of recursion
+//
+// Returns:
+//
+//	a single-dimensional list of elements from the input slice or array.
 func (sr *SlicesRegistry) flattenSlice(val reflect.Value, remainingDeep int) []any {
```

Comment:
flattenSlice uses `v` arg, but isComparable use val.

But then, Default and Empty uses `given` in next file

Maybe you could rename the art, and so update the doc

Please note, there might have more to check

Reply:
I will harmonize all params